core/src/main/java/org/elasticsearch/indices/recovery/RecoveryTargetService.java

```diff
@@ -263,12 +263,6 @@ public RecoveryResponse newInstance() {
                 return;
             }
 
-            if (cause instanceof IndexShardClosedException) {
```

Comment:
I wonder if we should move it up above the IllegalIndexShardStateException, as we don't want to retry in this case (although it's not that bad).